auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java

```diff
@@ -260,4 +267,34 @@ public String getRefreshToken() {
         return refreshToken;
     }
 
+
+    /**
+     * Returns the custom attribute.
+     *
+     * @param attribute the attribute name to retrieve
+     * @return the custom attribute.
+     */
+    @Override
+    public String getCustomAttribute(final String attribute) {
+        String attributeValue = "";
+        try {
+            String identityToken = getIdentityToken();
```

Comment:
Is this identity token same for the whole session? Maybe we shouldn't parse it every time, but cache the result.

Reply:
Hmm, I'm not sure if there is a need to cache it as we are really only Base64 decoding and parsing it as a JSON object.

Reply:
I can live with it, as it won't be used to exchange large amounts of data.